### ISSUE TITLE
Rename `aws_config::Config` to aws_config::SharedConfig`

### ISSUE THREAD

**Issue by Velfi** [awslabs/aws-sdk-rust#406]
### Tell us about your request

Rename `aws_config::Config` to aws_config::SharedConfig`

### Tell us about the problem you're trying to solve.

Users struggle to understand the difference between a `Config` struct from a service SDK and a `Config` struct from `aws_config`. Renaming the type would make users less likely to think the `Config` structs are interchangeable.

### Are you currently working around this issue?

It's possible to alias the type but this doesn't help users that don't already know the difference

**Comment by benesch** [comment]
Yes, please! This was one of my primary sources of confusion when learning the SDK.

One note: `SharedConfig` is easy to misinterpret. The correct interpretation is “configuration common to all SDK clients” (right?) but it’s reminiscent of a “global shared configuration object” like an Arc.

`CommonConfig` strikes me as maybe a bit clearer?

I definitely retract my suggestion if there’s prior art in other SDKs that use the “Shared” terminology, though.

**Comment by Velfi** [comment]
[This is now in PR](https://github.com/awslabs/smithy-rs/pull/1241)

**Comment by jdisanti** [comment]
This was released with [`v0.9.0`](https://github.com/awslabs/aws-sdk-rust/releases/tag/v0.9.0).
